### PR TITLE
Build native libs for linux-musl-arm and linux-musl-arm64

### PR DESCRIPTION
**Pull request by arturcic**
See #130 for initial PR.

I commented on the changes on the original PR. I think we need to continue the discussion here. 
This PR is made from a branch instead of `master`

### CODE REVIEW

**Review by bording — CHANGES_REQUESTED** [review]

**Comment by bording** [comment]
I did also merge #132 which has introduced a conflict here, sorry about that!

**Comment by arturcic** [comment]
> I did also merge #132 which has introduced a conflict here, sorry about that!

I'm glad that that one was merged. I'll rebase my changes and will work on changes you requested

File: build.libgit2.sh [hunk 1]
```diff
@@ -18,39 +17,20 @@ rm -rf libgit2/build
...
-if [[ $RID == *arm ]]; then
-	export TOOLCHAIN_FILE=/nativebinaries/CMakeLists.arm.txt
-fi
-
-if [[ $RID == *arm64 ]]; then
-	export TOOLCHAIN_FILE=/nativebinaries/CMakeLists.arm64.txt
-fi
-
 cmake -DCMAKE_BUILD_TYPE:STRING=Release \
+      -DCMAKE_BUILD_PARALLEL_LEVEL=16 \
```
[the remark this answers is not in the document]
Comment (arturcic) [reply build.libgit2.sh:21]: I was trying to improve the build time by running in parallel if possible. For this PR I will remove this

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -86,3 +93,7 @@ jobs:
         with:
           name: NuGet package
           path: ./*.nupkg
+      - name: Publish nuget to GitHub Packages
```
[the remark this answers is not in the document]
Comment (arturcic) [reply .github/workflows/ci.yml:96]: ok, I will remove this

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -47,6 +51,9 @@ jobs:
       - name: Build macOS
         if: runner.os == 'macOS'
         run: ./build.libgit2.sh
+      - name: Setup QEMU
```
[the remark this answers is not in the document]
Comment (arturcic) [reply .github/workflows/ci.yml:54]: I agree it's not nice

File: build.libgit2.sh [hunk 4]
```diff
@@ -18,39 +17,20 @@ rm -rf libgit2/build
...
       -DUSE_HTTPS=$USEHTTPS \
       -DUSE_BUNDLED_ZLIB=ON \
-      -DCMAKE_TOOLCHAIN_FILE=${TOOLCHAIN_FILE} \
       ..
-cmake --build .
+cmake --build . 
 
 popd
 
-if [[ $RID == "" ]]; then
```
[the remark this answers is not in the document]
Comment (arturcic) [reply build.libgit2.sh]: Revert the changes in this section

File: Dockerfile.linux [hunk 5]
```diff
@@ -1,7 +1,8 @@
-FROM alpine:3.7
+FROM multiarch/crossbuild
```
[the remark this answers is not in the document]
Comment (arturcic) [reply Dockerfile.linux:1]: Include `pkg-config` installation

**Comment by arturcic** [comment]
@bording rebased and fixed according change requests.

File: build.libgit2.sh [hunk 6]
```diff
@@ -5,7 +5,6 @@ set -e
 LIBGIT2SHA=`cat ./nuget.package/libgit2/libgit2_hash.txt`
 SHORTSHA=${LIBGIT2SHA:0:7}
 OS=`uname`
-ARCH=`uname -m`
```
Comment (bording) [inline build.libgit2.sh:8]: This line needs to come back also

File: build.libgit2.sh [hunk 7]
```diff
@@ -22,18 +21,6 @@ rm -rf libgit2/build
 mkdir libgit2/build
 pushd libgit2/build
 
-export _BINPATH=`pwd`
```
Comment (bording) [inline build.libgit2.sh:25]: This line should also come back.

**Review by bording — CHANGES_REQUESTED** [review]

File: build.libgit2.sh (hunk 6 above)
Comment (arturcic) [reply build.libgit2.sh:8]: right

Comment (arturcic) [reply build.libgit2.sh]: done

File: build.libgit2.sh (hunk 7 above)
Comment (arturcic) [reply build.libgit2.sh:25]: done

**Review by bording — APPROVED** [review]

**Comment by arturcic** [comment]
@bording thank you for merging this one. Do you have any plans to release a new version of the package?

**Comment by bording** [comment]
> @bording thank you for merging this one. Do you have any plans to release a new version of the package?

Yep, just waiting for the linux-musl builds to finish, they are so slow!

**Comment by arturcic** [comment]
Agree, one of the reasons I wanted to run in parallel. Any plans also to release a LibGit.Sharp new version to reference the new native package?

**Comment by bording** [comment]
> Agree, one of the reasons I wanted to run in parallel.

I looked at the build times between CI runs with and without the parallel setting, and it wasn't a clear win. Some things got a little bit faster, but the only things that really matter are the musl-arm builds.

The `linux-musl-arm64` build does see some big improvements with the parallel setting, but the `linux-musl-arm` build appears to get slower with the setting, removing any gains and making the total build time about the same.

>  Any plans also to release a LibGit.Sharp new version to reference the new native package?

I won't be able to get to that today, so if you wanted to open a PR that included an update to the new native binaries package and introduced some additional linux distros into the testing matrix, that would be appreciated.

**Comment by arturcic** [comment]
> I won't be able to get to that today, so if you wanted to open a PR that included an update to the new native binaries package and introduced some additional linux distros into the testing matrix, that would be appreciated.

I will have a look

**Comment by bording** [comment]
2.0.315-alpha.0.7 has been pushed to nuget.org.

**Pull request merged**